base/range.jl

```diff
@@ -866,7 +866,7 @@ function getindex(r::AbstractUnitRange, s::StepRange{T}) where {T<:Integer}
                 return range(first(r), step=one(eltype(r)), length=0)
             end
         else # length(s) == 2
-            return range(r[2], step=one(eltype(r)), length=1)
+            return range(last(r), step=one(eltype(r)), length=1)
```

Comment:
when reviewing this PR I was not fully clear if we should not use `oneunit(eltype(r))` instead of `one(eltype(r))` everywhere in these codes. What do you think?

Reply:
`oneunit` seems to be clearer in intent

Reply:
OK - changed. I also used the optimization you have suggested above (but could you please double check it please?). Thank you!